[21.05] devhost: replace hydra eval with metadata endpoint, various improvements (PL-132065)

PhilTaken opened the pull request:
@flyingcircusio/release-managers

relates to [PL-132065]

## Release process

Impact:

This PR implements a new subcommand "provision" for `fc-devhost` that takes a channel url and an image url instead of a hydra eval (see subcommand "ensure") to provision a new devhost vm. Functionally, the new subcommand is identical to the existing subcommand "ensure".
Additionally, it changes the destroy subcommand to accept an arbitrary number of vm names that will all be deleted.
The `destroy` subcommand is also aliased to `rm` similarly to `list` being aliased to `ls`.

Changelog:

- implement a new subcommand "provision" for `fc-devhost` that takes a channel url and an image url instead of a hydra eval (see subcommand "ensure") to provision a new devhost vm
- change the `destroy` subcommand to take an arbitrary (n >= 1) amount of vm names to destroy
- set the program name in `argparse` for more readable error messages that don't include a nix store hash

### PR release workflow (internal)

- [x] PR has internal ticket
- [x] internal issue ID (PL-…) part of branch name
- [x] internal issue ID mentioned in PR description text
- [x] ticket is on Platform agile board
- [ ] ticket state set to *Pull request ready*
- [ ] if ticket is more urgent than within the next few days, directly contact a member of the Platform team

## Design notes

- [ ] Provide a feature toggle if the change might need to be adjusted/reverted quickly depending on context. Consider whether the default should be `on` or `off`. Example: rate limiting.
- [ ] All customer-facing features and (NixOS) options need to be discoverable from documentation. Add or update relevant documentation such that hosted and guided customers can understand it as well.

## Security implications

- [x] [Security requirements](https://wiki.flyingcircus.io/System_Development_Guideline#Security_requirement_principles_and_testing) defined? (WHERE) 

We now allow users to provide explicit image URLs to bootstrap VMs from. Theoretically those could be arbitrary OS images including Windows, memtest, or some malware that then get access to the internet via NAT.

However, the VM architecture itself already provides similar vectors, like becoming root and spawning custom kernels. This is the whole point of running VMs anyway.

- [x] Security requirements tested? (EVIDENCE)

Functionality tested manually with @leona-ya on largo00.

PhilTaken committed:
fc-devhost: provison subcommand

- takes an image + a channel url instead of a hydra eval to provision
  devhosts

PhilTaken committed:
fc-devhost: delete multiple vms in one invocation

- also adds rm alias for destroy

PhilTaken committed:
fc-devhost: set program name

leona-ya commented:
Blocked by https://gitlab.flyingcircus.io/flyingcircus/fc.directory/-/merge_requests/377

ctheune committed:
devhost: clean up integration of hydra_eval vs. explicit urls

We stay backwards compatible and have enhanced a few minor aspects.

Re PL-132065
